crowdfunding/projects/serializers.py

```diff
@@ -0,0 +1,63 @@
+from rest_framework import serializers
+
+from .models import Project, Pledge
+from users.serializers import CustomUserSerializer
+
+class PledgeSerializer(serializers.ModelSerializer):
+    supporter = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Pledge
+        fields = ['id', 'amount', 'comment', 'anonymous', 'project', 'supporter']
+        read_only_fields = ['id', 'supporter']
+
+    def get_supporter(self, instance):
+        
+        if instance.anonymous:
+            return "The Lego Elves!"
+        else:
+            return instance.supporter.username
+
+class PledgeDetailSerializer(PledgeSerializer):
+
+    def update(self, instance, validated_data):
+        instance.amount = validated_data.get('amount', instance.amount)
+        instance.comment = validated_data.get('comment', instance.comment)
+        instance.anonymous = validated_data.get('anonymous', instance.anonymous)
+        instance.supporter = validated_data.get('supporter',instance.supporter)
+        instance.project = validated_data.get('project', instance.project)
+        instance.save()
+        return instance
```

Comment:
i'm not sure you need this class, as the update process should be handled by default by the model serializer